src/libAtomVM/term.h

```diff
@@ -51,8 +51,8 @@
 #define TERM_BOXED_REFC_BINARY 0x20
 #define TERM_BOXED_HEAP_BINARY 0x24
 
-#define TERM_BOXED_REFC_BINARY_SIZE 3
-#define TERM_BOXED_BIN_MATCH_STATE_SIZE 4
+#define TERM_BOXED_REFC_BINARY_SIZE 6
+#define TERM_BOXED_BIN_MATCH_STATE_SIZE 6
```

Comment:
we should find a way to save precious bytes.

Reply:
Per the above, AFAIK the creation of a match state requires incrementing the reference count when it is created, and decrementing it when it is GC'd, just like a copy from a process to a mailbox, and conversely.  I would be happy to be proved wrong about this.

Reply:
Fixed.  Reverted changed to match states.